apps/designer/app/canvas/shared/styles.ts

```diff
@@ -97,11 +94,28 @@ export const GlobalStyles = ({ assets }: { assets: Array<Asset> }) => {
       tokensCssEngine.addStyleRule(`:root`, { style });
     }
 
-    if (typeof document !== "undefined") {
-      tokensCssEngine.render();
-    }
+    tokensCssEngine.render();
   }, [tokens]);
 
+  const presetStyle = usePresetStyle();
+
+  useIsomorphicLayoutEffect(() => {
+    presetStyleEngine.clear();
+    const rules = new Map<string, StyleRule>();
+    for (const { component, property, value } of presetStyle) {
+      let rule = rules.get(component);
+      if (rule === undefined) {
+        rule = presetStyleEngine.addStyleRule(
```

Comment:
as an alternative to adding style rule for preset, we could be merging styles with the local styles, do you think this approach is better?

Reply:
It adds more work on every style update. Here it is done only once. Do you see any downside with using data-ws-component selector?

Reply:
only that I was hoping to delete all data attributes except of data-ws-id, but if this use case is super important we could do it, I am going to read more and test soon